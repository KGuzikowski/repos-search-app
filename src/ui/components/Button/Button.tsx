import styled from 'styled-components'
import { primaryColor, secondaryColor } from '../../style-variables'

const reversed = `
    background: ${primaryColor};
    color: ${secondaryColor};
    border: 2px solid ${primaryColor};
`

const normal = `
    background: ${secondaryColor};
    color: ${primaryColor};
    border: 1px solid ${secondaryColor};
`

const Button = styled.a<{reversed?: boolean}>`
    text-decoration: none;
    font-size: 15px;
    padding: 10px 20px;
    cursor: pointer;
    font-weight: bold;
    border-radius: 5px;
    transition: color, background 200ms ease-in;

    ${props => props.reversed ? reversed : normal}

    &:hover {
        color: ${props => props.reversed ? primaryColor : '#fff'};
        background: transparent;
    }
`

export default Button