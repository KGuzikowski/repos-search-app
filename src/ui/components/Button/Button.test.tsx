import React from 'react'
import { create } from 'react-test-renderer'
import Button from './Button'

it('renders Button component correctly', () => {
    const component = create(<Button>Testing</Button>)
    expect(component.toJSON()).toMatchSnapshot()
})