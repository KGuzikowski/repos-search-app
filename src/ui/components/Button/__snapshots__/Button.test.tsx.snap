// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component correctly 1`] = `
<a
  className="sc-AxjAm bCzWPK"
>
  Testing
</a>
`;
